### PR TITLE
restore redis interface

### PR DESCRIPTION
**Pull request by wangxicoding**
非常感谢您给EDL项目提交PR!  
在提交PR之前，请帮忙回答一下问题以帮助我们判断PR的意图：  
- 这个PR要解决什么问题？
- 有没有对应的ISSUE？ 
  Fix
- 这个PR里边有没有要注意的地方？
- 这个PR改变了用户接口？
- 其他的说明？

You are welcome to submmit PR for EDL.   
Before this, would you like to answer some questions to help others to get what the PR does? 
- What this PR does / why we need it?
- Which issue(s) this PR fixes? 
  Fix
- Special notes for your reviewer:
- Does this PR introduce a user-facing change?
- Additional documentation?

### CODE REVIEW

File: python/paddle_edl/distill/distill_reader.py [hunk 1]
```diff
@@ -52,9 +52,15 @@ def __init__(self, discovery_servers, require_num, service_name):
         self._require_num = require_num
         self._service_name = service_name
         self._client = None
+        self._type = os.environ.get('PADDLE_DISTILL_BALANCE_TYPE', 'redis')
 
     def _connect(self):
-        from paddle_edl.distill.discovery_client import DiscoveryClient
+        if self._type == 'etcd':
+            from paddle_edl.distill.discovery_client import DiscoveryClient
```
Comment (gongweibao) [inline python/paddle_edl/distill/distill_reader.py:59]: 现在是否需要assert 非etcd?

File: example/distill/mnist_distill/run_student.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -eu
+
+# paddle cloud will set discovery env, set for local test
+export PADDLE_DISTILL_BALANCE_SERVER='10.255.100.13:9379'
+#export PADDLE_DISTILL_BALANCE_SERVER='127.0.0.1:9379'
```
Comment (gongweibao) [inline example/distill/mnist_distill/run_student.sh]: clean?

File: example/distill/mnist_distill/train_student_with_fleet.py [hunk 3]
```diff
@@ -106,16 +106,19 @@ def train(nn_type,
...
-            BATCH_SIZE,
-            d_batch_size=8,
-            capacity=4,
-            occupied_capacity=2)
+            ins=['img', 'label'],
+            predicts=['prediction'],
+            conf_file='mnist_client_conf/serving_client_conf.prototxt')
+        dr.set_teacher_batch_size(16)
+
+        #dr.set_fixed_teacher(['127.0.0.1:9292', '127.0.0.1:9293'])
```
Comment (gongweibao) [inline example/distill/mnist_distill/train_student_with_fleet.py]: clean?

File: example/distill/mnist_distill/train_student_with_fleet.py [hunk 4]
```diff
@@ -106,16 +106,19 @@ def train(nn_type,
...
-        assert BATCH_SIZE % 8 == 0
         dr = DistillReader(
-            'mnist_client_conf/distill_reader.conf',
-            BATCH_SIZE,
-            d_batch_size=8,
-            capacity=4,
-            occupied_capacity=2)
+            ins=['img', 'label'],
+            predicts=['prediction'],
+            conf_file='mnist_client_conf/serving_client_conf.prototxt')
```
Comment (gongweibao) [inline example/distill/mnist_distill/train_student_with_fleet.py]: 还有conf?

**Review by gongweibao — CHANGES_REQUESTED** [review]

File: example/distill/mnist_distill/run_student.sh (hunk 2 above)
Comment (wangxicoding) [reply example/distill/mnist_distill/run_student.sh]: done

File: example/distill/mnist_distill/train_student_with_fleet.py (hunk 3 above)
Comment (wangxicoding) [reply example/distill/mnist_distill/train_student_with_fleet.py]: done

File: example/distill/mnist_distill/train_student_with_fleet.py (hunk 4 above)
Comment (wangxicoding) [reply example/distill/mnist_distill/train_student_with_fleet.py]: done. 下载功能稍后添加

**Review by gongweibao — APPROVED** [review]
LGTM

**Pull request merged**